proposals/20210126-nc-and-ncp.md

```diff
@@ -0,0 +1,218 @@
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+}
+
+type NestedClusterStatus struct {
+	// Ready is when the NestedControlPlane has a API server URL.
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+}
+
+type NestedCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NestedClusterSpec   `json:"spec,omitempty"`
+	Status NestedClusterStatus `json:"status,omitempty"`
+}
+```
+
+#### NestedControlPlane
+
+Kubernetes API Group: `controlplane.cluster.x-k8s.io/v1alpha4`
+
+The NestedControlPlane resource is responsible for orchestrating the overarching cluster, this controller doesn't create any of the downstream objects instead it creates a place where the downstream objects can look up shared values. 
+
+**Question:**
+1. should we have information necessary for the syncer within this object? 
+
+```go=
+type NestedControlPlaneSpec struct {
+	// NamespacePrefix is the namespace where the control plane is deployed as well as the prefix ++ tenant control plane namespace name for all workloads 
+	// +optional
+	NamespacePrefix string `json:"namespacePrefix,omitempty"`
+    
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+    
+	// EtcdRef is the reference to the NestedEtcd 
+	EtcdRef *corev1.ObjectReference `json:"etcd,omitempty"` 
+    
+	// APIServerRef is the reference to the NestedAPIServer 
+	// +optional
+	APIServerRef *corev1.ObjectReference `json:"apiserver,omitempty"` 
+    
+	// ContollerManagerRef is the reference to the NestedControllerManager
+	// +optional
+	ControllerManagerRef *corev1.ObjectReference `json:"controllerManager,omitempty"`
+}
+
+type NestedControlPlaneStatus struct {
+	// ExternalManagedControlPlane indicates to cluster-api that the control plane
+	// is managed by an external service such as AKS, EKS, GKE, etc.
+	// +kubebuilder:default=true
+	ExternalManagedControlPlane *bool `json:"externalManagedControlPlane,omitempty"`
+    
+	// Initialized denotes whether or not the control plane has the
+	// uploaded kubernetes config-map.
+	// +optional
+	Initialized bool `json:"initialized"`
+    
+	// Ready denotes that the AWSManagedControlPlane API Server is ready to
+	// receive requests and that the VPC infra is ready.
+	// +kubebuilder:default=false
+	Ready bool `json:"ready"`
+    
+	// ErrorMessage indicates that there is a terminal problem reconciling the
+	// state, and will be set to a descriptive error message.
+	// +optional
+	FailureMessage *string `json:"failureMessage,omitempty"`
+    
+	// Conditions specifies the cpnditions for the managed control plane
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+}
+```
+
+
+### Requirements (Optional)
+
+Each NestedControlPlane MUST have `ExternalManagedControlPlane` set to true.
```

Comment:
What if the control plane is created natively on the super cluster?

Reply:
Yea, I read some capi requirements wrong this shouldn't be required.